addons/website_profile/controllers/main.py

```diff
@@ -235,13 +237,13 @@ def view_all_users_page(self, page=1, **kwargs):
             users = User.sudo().search(dom, limit=self._users_per_page, offset=pager['offset'], order='karma DESC')
             user_values = self._prepare_all_users_values(users)
 
+            current_user = User.sudo().search(expression.AND([[('id', '=', uid)], dom]))
+            if (current_user not in users):
```

Comment:
so you add first the sql query, and check after if it was useless ???

did you mean:
```
if uid not in users.ids:
   current_user_values = self._prepare_all_users_values(request.env.user)

```
??

Please try to optimize your code
We are not paid according the number of sql queries or python line of code.